components/calendar/calendar.js

```diff
@@ -325,57 +331,66 @@ class Calendar extends LocalizeStaticMixin(LitElement) {
 
 		const dates = getDatesInMonthArray(this._shownMonth, this._shownYear);
 		const dayRows = dates.map((week) => {
-			const weekHtml = week.map((day) => {
+			const weekHtml = week.map((day, index) => {
+				const focused = checkIfDatesEqual(day, this._focusDate);
+				const selected = this.selectedValue ? checkIfDatesEqual(day, parseISODate(this.selectedValue)) : false;
 				const classes = {
 					'd2l-calendar-date': true,
-					'd2l-calendar-date-selected': this.selectedValue ? checkIfDatesEqual(day, parseISODate(this.selectedValue)) : false,
+					'd2l-calendar-date-selected': selected,
 					'd2l-calendar-date-today': checkIfDatesEqual(day, this._today)
 				};
-				const focused = checkIfDatesEqual(day, this._focusDate);
+				const year = day.getFullYear();
+				const month = day.getMonth();
+				const date = day.getDate();
 				return html`
 					<td>
 						<div
+							aria-label="${formatDate(day, {format: 'full'})}"
+							aria-selected="${selected}"
 							@click="${this._onDateSelected}"
 							@keydown="${this._onKeyDown}"
 							class=${classMap(classes)}
-							data-date=${day.getDate()}
-							data-month=${day.getMonth()}
-							data-year=${day.getFullYear()}
+							data-date=${date}
+							data-month=${month}
+							data-year=${year}
+							headers="${this._tableInfoId}-weekday-${index}"
```

Comment:
Typically just having things in standard table markup gets you accessible header handling for free, so I'm curious why this is necessary. Might moving the role/aria-* stuff to the `<td>` fix things?

Reply:
I was thinking of trying to restructure this part to do that so I'll give a stab at it.